test-monitor/nitestmonitor.yml

```diff
@@ -0,0 +1,1277 @@
+        items:
+          type: string
+        example:
+          - 9c75f1b5-5882-490e-a2e2-1c14123d68b2
+      resultIds:
+        description: An array of result ids
+        type: array
+        items:
+          type: string
+        example:
+          - 9c75f1b5-5882-490e-a2e2-1c14123d68b2
+      path:
+        description: A step path
+        type: string
+        example: root.step1.step2
+      pathIds:
+        description: An array of path ids
+        type: array
+        items:
+          type: string
+        example:
+          - 9c75f1b5-5882-490e-a2e2-1c14123d68b2
+          - 1f05c3c8-7a8c-4ebb-a87f-7a60d5dd8938
+      statuses:
+        description: An array of step statuses
+        type: array
+        items:
+          $ref: '#/definitions/StatusObject'
+      startedAtQuery:
+        description: A time query object
+        type: array
+        items:
+          $ref: '#/definitions/TimeQueryObject'
+      updatedAtQuery:
+        description: A time query object
+        type: array
+        items:
+          $ref: '#/definitions/TimeQueryObject'
+      sortBy:
+        type: array
+        items:
+          $ref: '#/definitions/StepSortDefinitionObject'
+  TestStepRequestObject:
+    description: A test step object
+    type: object
+    properties:
+      stepId:
+        description: The step id
+        type: string
+        example: Step1
+      parentId:
+        description: The parent step id
+        type: string
+        example: Step0
+      resultId:
+        description: The result id
+        type: string
+        example: 1f05c3c8-7a8c-4ebb-a87f-7a60d5dd8938
+      children:
+        description: Nested child steps
+        type: array
+        items:
+          $ref: '#/definitions/TestStepRequestObject'
+      data:
+        $ref: '#/definitions/StepDataObject'
+      dataModel:
+        description: The step data model
+        type: string
+        example: TestStand
+      name:
+        description: The step name
+        type: string
+        example: My Step
+      startedAt:
+        description: An ISO-8601 formatted timestamp indicating when the result was started
+        type: string
+        format: iso-date-time
+        example: '2018-05-07T18:58:05.219692Z'
+      status:
+        $ref: '#/definitions/StatusObject'
+      stepType:
+        description: The step type
+        type: string
+        example: NumericLimitTest
+      totalTimeInSeconds:
+        description: The total number of seconds the step took to execute
+        type: number
+        format: double
+        default: 0
+        example: 29.9
+  TestStepCreateOrUpdateRequestObject:
+    description: A test step request object
+    type: object
+    required:
+      - steps
+    properties:
+      steps:
+        description: An array of test steps to create
+        type: array
+        items:
+          $ref: '#/definitions/TestStepRequestObject'
+  TestStepResponseObject:
+    description: A test step response object
```

Comment:
Same note about adding value for less HTTP API-savvy users by rephrasing and/or adding (very brief) context